### DISCUSSION TITLE
how to username for traces in span attribute

### DISCUSSION THREAD

**Discussion by raagulrakesh** [grafana/faro-web-sdk#537] [General]
Hi,
I have enabled faro for traces and logs. I am stuck in sending username in traces.  In log able to send username by using faro.api.setuser.
Is it possible to send username for all traces as span attributes

**Comment by codecapitano** [comment]
Hi @raagulrakesh,
Currently there's no built-in support to attach user data to traces/spans.
The only solution atm is to use otel APIs to add it manually unfortunately.

I opened an issue to add support for user attributes to the web-tracing package
https://github.com/grafana/faro-web-sdk/issues/540

Cheers,
Marco

**Reply by raagulrakesh** [reply]
Thanks for the support @codecapitano
